src/wfdirsrc.c

```diff
@@ -1046,8 +1046,16 @@ DSDropObject(
    // so, don't do anything.
    //
    if (hwndHolder == lpds->hwndSource) {
-      if ((dwSelSink == (DWORD)-1) || SendMessage(hwndLB, LB_GETSEL, dwSelSink, 0L))
-         return TRUE;
+      if ((dwSelSink == (DWORD)-1) || SendMessage(hwndLB, LB_GETSEL, dwSelSink, 0L)) {

```

Comment:
The second part of this test means that if a user Ctrl+drags a file over itself, it will be copied; but Ctrl+drag over any other file will do nothing.  Is this intended?

Note that because this is checking the list box hwnd, dragging between two windows to the same directory have different behavior here: Ctrl+drag to a different file in the same directory but a second window will copy (adding the new suffix.)

It looks like this section can't interpret dragging over other files as copies without breaking the code below which is attempting to launch a program with a file as an argument.

I think this check used to be at the top to just short circuit logic.  Note the "NormalMoveCopy" case comments that it's checking for the same directory, but really doesn't - it's checking for the same window, so the previous code would attempt the operation when crossing windows and fail later.

I'm wondering if this check should really look at fShowSourceBitmaps to no-op moves, but allow copies to continue; and the "NormalMoveCopy" case should also look at fShowSourceBitmaps to avoid the same-window check for copies, which this PR is aiming to support.

Reply:
Investigating ....

Reply:
Did you try this or just guessed from the code?

I just tried this.
If I drag a file over an .exe it does start the .exe with the file as command line argument, regardless if it is in the same window or in a different one.
If I drag a file on the empty space it does a copy with the '- Copy' postfix.

Does this resolve your concern?
Do you have a use-case where you observed that it did not work?

Reply:
I tried it.  I saw the same behavior for both cases you listed.  The issue is what happens when files are dragged over each other (ie., not the empty space.)  The case that seemed interesting to me is Ctrl+drag a file over a different file, which is a no-op in the same window, but will generate a copy if dragged to a different window.  The behavior difference is really an artifact though of having slightly different pieces of code processing what is effectively the same case (by checking the window for equality), and I think removing the duplication will result in a more consistent experience.

Reply:
My current summary on this thread so far is
### Copy over files over 'ops' (.exe/.bat/...)
- Works as before

### Copy over no-op over no-ops

- Code of this PR
  same pane: nothing happens
  different pane: copy is created with '- Copy' postfix
  
https://user-images.githubusercontent.com/7418603/167071945-17aaa0e9-e287-4357-9585-f61f5f20d8d7.mp4


  
- Previous Code  
  same pane: nothing happens
  different pane: Box pops up that files are the same <the video grabber didn't capture this>

https://user-images.githubusercontent.com/7418603/167071968-f11af98c-bb93-4b24-9c3a-3bdca53472b0.mp4

![messagebox](https://user-images.githubusercontent.com/7418603/167072049-3b1b7523-7fa4-4490-9ac8-34be057fe85e.png)

  
So I don't see a change of behaviour, except for the '- copy' postfix.
I don't want to change the same-window check in this PR. I just wanted to create postfixes.

Did I miss something? Any breaking behaviour change?